### PR TITLE
Decoupling model from repository implementation

### PR DESCRIPTION
**Pull request by andresmoschini**
Hi team!

Previously our `ContentRow` was tied to `Repository`implementation.

Now, the repository implementation is responsible to understand Doppler data and create the right `ContentData` to represent our contents.

I want to include repository implementation code in our integration tests, but it requires adding types to query results (I am using dynamics results now), so, by the moment I have only added TODO notes.

These are the different `ContentData` that we know now:

image

And, with this diagram I am trying to represent the logic behind the namespaces and folder structure in the project:

image

(I am not sure if this component diagram is fine, I am not so familiar with this kind of diagrams)

### CODE REVIEW

File: Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    public Repository(IDatabaseConnectionFactory connectionFactory)
+    {
+        _connectionFactory = connectionFactory;
+    }
+
+    public async Task<ContentData> GetCampaignModel(string accountName, int campaignId)
+    {
+        using (var connection = await _connectionFactory.GetConnection())
+        {
+            var databaseQuery = @"
```
Comment (ms-darianbenito) [inline Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs:25]: just a detail, we can use a constant here
```suggestion
            const string databaseQuery = @"
```

**Comment by ms-darianbenito** [comment]
Good job! thanks for sharing 👍

File: Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            // TODO: test this scenario
+            if (!campaignStatus.OwnCampaignExists)
+            {
+                throw new ApplicationException($"CampaignId {contentRow.campaignId} does not exists or belongs to another user than {accountName}");
+            }
+
+            // TODO: test these scenarios
+            var query = campaignStatus.ContentExists
+                ? @"UPDATE Content SET Content = @Content, Meta = @Meta, EditorType = @EditorType WHERE IdCampaign = @IdCampaign"
+                : @"INSERT INTO Content (IdCampaign, Content, Meta, EditorType) VALUES (@IdCampaign, @Content, @Meta, @EditorType)";
```
Comment (ms-darianbenito) [inline Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs:120]: I think that It is not necessary 
```suggestion
                ? "UPDATE Content SET Content = @Content, Meta = @Meta, EditorType = @EditorType WHERE IdCampaign = @IdCampaign"
                : "INSERT INTO Content (IdCampaign, Content, Meta, EditorType) VALUES (@IdCampaign, @Content, @Meta, @EditorType)";
```

**Review by maurocardinali — COMMENTED** [review]
Reviewed 🤓

File: Doppler.HtmlEditorApi/Storage.DapperProvider/DatabaseConnectionFactory.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+
+namespace Doppler.HtmlEditorApi.Storage.DapperProvider;
+
+public class DatabaseConnectionFactory : IDatabaseConnectionFactory
+{
+    private readonly string _connectionString;
+
+    public DatabaseConnectionFactory(IOptions<DopplerDatabaseSettings> dopplerDataBaseSettings)
```
[the remark this answers is not in the document]
Comment (andresmoschini) [reply Doppler.HtmlEditorApi/Storage.DapperProvider/DatabaseConnectionFactory.cs:13]: Done here: https://github.com/FromDoppler/doppler-html-editor-api/pull/42/files#diff-9d3c2ee22baed3f37ff7097f9988e3ee107467e2153927d5fb2ba3f22b851910R13

File: Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs (hunk 1 above)
Comment (andresmoschini) [reply Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs:25]: Done here: https://github.com/FromDoppler/doppler-html-editor-api/pull/44/files#diff-670f8ae2f5aff6be803837e52585a38c81e9738fa7b599d982d52a2836c64f36R9

**Pull request merged**